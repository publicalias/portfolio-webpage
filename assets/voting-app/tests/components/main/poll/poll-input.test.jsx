"use strict";

//local imports

const PollInput = require("../../../../scripts/components/main/poll/poll-input");

const { initTestPoll, testReload, testWrapper } = require("../../../test-helpers");

//global imports

const { testMock } = require("test-helpers/meta-tests");
const { initTestSnapshot, reactTests } = require("test-helpers/react-tests");

//utilities

const testLoad = (render, args) => {

  const { submitKeys } = Object.assign(PollInput.injected, { submitKeys: jest.fn() });

  const { wrapper } = render();

  wrapper.mount();

  testMock(submitKeys, args);

  wrapper.unmount();

};

const testChange = (render, type) => {

  const { props, wrapper } = render();

  wrapper.find(".qa-option-input").simulate("change", { target: { value: "Option A" } });

  testMock(props.actions[type], ["Option A"]);

  wrapper.unmount();

};

//setup

beforeAll(reactTests.setup);

beforeEach(() => {
  PollInput.injected.submitKeys = jest.fn();
});

//poll input

describe("poll input (form)", () => {

  const { testMount, testShallow } = testWrapper(PollInput);

  const testSnapshot = initTestSnapshot(testShallow);
  const testForm = initTestPoll(testSnapshot, "form");
  const testFormMount = initTestPoll(testMount, "form");

  it("should match snapshot (default)", () => testForm());

  it("should match snapshot (add)", () => testForm({ form: { add: "Option A" } }));

  it("should call submitKeys on load", () => testLoad(testFormMount, ["form"]));

  it("should call formSetAdd on change", () => testChange(testFormMount, "formSetAdd"));

  it("should call formAddOption and formSetAdd on submit", () => {

    const { props, wrapper } = testFormMount({ form: { add: "Option A" } });

    const { actions: { formAddOption, formSetAdd } } = props;

    wrapper.find(".qa-option-submit").simulate("click");

    testMock(formAddOption, []);
    testMock(formSetAdd, [""]);

    wrapper.unmount();

  });

});

describe("poll input (view)", () => {

  const { testMount, testShallow } = testWrapper(PollInput);

  const testSnapshot = initTestSnapshot(testShallow);
  const testView = initTestPoll(testSnapshot, "view");
  const testViewMount = initTestPoll(testMount, "view");

  it("should match snapshot (default)", () => testView());

  it("should match snapshot (add)", () => testView({ view: { add: "Option A" } }));

  it("should call submitKeys on load", () => testLoad(testViewMount, ["view"]));

  it("should call viewSetAdd on change", () => testChange(testViewMount, "viewSetAdd"));

  it("should call pollAddOption and viewSetAdd on submit", async () => {

    const { props, wrapper } = testViewMount({ view: { add: "Option A" } }, { id: "id-a" });

    const { actions: { viewSetAdd } } = props;

    await testReload(props, wrapper, ".qa-option-submit", "pollAddOption", ["id-a", "Option A"]);

    testMock(viewSetAdd, [""]);

  });

});
