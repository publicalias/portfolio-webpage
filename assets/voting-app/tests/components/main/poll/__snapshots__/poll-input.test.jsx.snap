// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`poll input (form) should match snapshot (add) 1`] = `
<div
  className="c-poll-options__input-box"
>
  <input
    className="c-poll-options__input js-submit-input-form qa-option-input u-margin-right"
    maxLength="100"
    onChange={[Function]}
    placeholder="New Option"
    value="Option A"
  />
  <button
    className="c-poll-options__submit js-submit-button-form qa-option-submit"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;

exports[`poll input (form) should match snapshot (default) 1`] = `
<div
  className="c-poll-options__input-box"
>
  <input
    className="c-poll-options__input js-submit-input-form qa-option-input u-margin-right"
    maxLength="100"
    onChange={[Function]}
    placeholder="New Option"
    value=""
  />
  <button
    className="c-poll-options__submit js-submit-button-form qa-option-submit"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;

exports[`poll input (view) should match snapshot (add) 1`] = `
<div
  className="c-poll-options__input-box"
>
  <input
    className="c-poll-options__input js-submit-input-view qa-option-input u-margin-right"
    maxLength="100"
    onChange={[Function]}
    placeholder="New Option"
    value="Option A"
  />
  <button
    className="c-poll-options__submit js-submit-button-view qa-option-submit"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;

exports[`poll input (view) should match snapshot (default) 1`] = `
<div
  className="c-poll-options__input-box"
>
  <input
    className="c-poll-options__input js-submit-input-view qa-option-input u-margin-right"
    maxLength="100"
    onChange={[Function]}
    placeholder="New Option"
    value=""
  />
  <button
    className="c-poll-options__submit js-submit-button-view qa-option-submit"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
